test/integration/asyncprofiler_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/parca-dev/parca-agent/pkg/asyncprofiler"
+	"github.com/parca-dev/parca-agent/pkg/convert"
+)
+
+func TestIntegrationAsyncProfiler(t *testing.T) {
+	cwd, err := os.Getwd()
+	require.NoError(t, err, "Failed to get current working directory")
+	// Start the Java program
+	javaProgramPath := filepath.Join(cwd, "testdata", "java-app", "target", "first-application-0.0.1-SNAPSHOT.jar")
+	cmd := exec.Command("java", "-Xms1G", "-Xmx1G", "-XX:+AlwaysPreTouch", "-jar", javaProgramPath)
+	fmt.Fprintf(os.Stdout, "Executing command: %v\n", cmd)
+	err = cmd.Start()
+	require.NoError(t, err, "Failed to start Java program")
+
+	time.Sleep(50 * time.Second) // We need to wait for the Java program to start
+	// Get the Java process PID
+	javaPID := cmd.Process.Pid
+	fmt.Fprintf(os.Stdout, "Java PID: %d\n", javaPID)
+
+	// Initialize AsyncProfiler with required parameters
+	profiler := asyncprofiler.NewAsyncProfiler(
+		filepath.Join(cwd, "testdata", "jattach"),
+		filepath.Join(cwd, "testdata", "libasyncProfiler.so"),
+		javaPID,
+	)
+
+	// Run fdtransfer
+	err = asyncprofiler.RunFdtransfer(filepath.Join(cwd, "testdata", "fdtransfer"), javaPID)
```

Comment:
I thought we do the `sysctl` so we don't need to do this?

Reply:
sysctl is only used for now. We do want this to work, right? I wanted to still keep this so that when fdtransfer related bug is fixed, all we need to do is remove sysctl from the github action. But if you want me to add this only when it works, I can remove it.